src/huggingface_hub/repocard_data.py

```diff
@@ -121,6 +121,17 @@ class EvalResult:
     # A JSON Web Token that is used to verify whether the metrics originate from Hugging Face's [evaluation service](https://huggingface.co/spaces/autoevaluate/model-evaluator) or not.
     verify_token: Optional[str] = None
 
+    @property
+    def unique_identifier(self) -> tuple:
+        """Returns a tuple that uniquely identifies this evaluation."""
+        return (
+            self.task_type,
+            self.dataset_type,
+            self.dataset_config,
+            self.dataset_split,
+            self.dataset_revision,
```

Comment:
I decided to exclude `dataset_args` from this tuple because it's a `dict` in general which causes hashing errors when we try to use it to access keys [here](https://github.com/huggingface/huggingface_hub/pull/1214/files#diff-647a41158f8522ca63688407c498ac771f2d77fae4e0af672ab29ef8567202afR530)

IMO `dataset_args` is unlikely to be used much for defining a model evaluation since the `config` and `split` etc capture the salient info

Reply:
Totally fine with me :+1: